### PR TITLE
Encapsulate ARguments in Named Tuples (Continued)

### PR DESCRIPTION
**Pull request by kolanos**
Related #132

### CODE REVIEW

**Review by mrickard — DISMISSED** [review]

**Review by mrickard — APPROVED** [review]

**Review by MattWhelan — APPROVED** [review]

File: newrelic_lambda_cli/cli/layers.py [hunk 1]
```diff
@@ -94,54 +95,24 @@ def register(group):
...
+    functions = get_aliased_functions(input)
 
     with ThreadPoolExecutor() as executor:
         futures = [
-            executor.submit(
-                config_layers,
-                function,
-            )
-            for function in functions
+            executor.submit(layers.install, input, function) for function in functions
```
Comment (MattWhelan) [inline newrelic_lambda_cli/cli/layers.py:115]: Wait, we got rid of the config_layers closure. It looks like we're not removing the subscription filter anymore.

**Review by MattWhelan — CHANGES_REQUESTED** [review]

File: newrelic_lambda_cli/layers.py [hunk 2]
```diff
@@ -206,17 +182,26 @@ def install(
...
-        if enable_extension and policy_arn:
+        if input.enable_extension and policy_arn:
             _attach_license_key_policy(
-                session, config["Configuration"]["Role"], policy_arn
+                input.session, config["Configuration"]["Role"], policy_arn
             )
-        if verbose:
+
+        if input.enable_extension_function_logs:
+            subscriptions.remove_log_subscription(input.session, function_arn)
```
Comment (MattWhelan) [inline newrelic_lambda_cli/layers.py:191]: Oh, nevermind. You moved that logic to here. That works too.

**Review by MattWhelan — APPROVED** [review]

File: newrelic_lambda_cli/cli/layers.py (hunk 1 above)
Comment (kolanos) [reply newrelic_lambda_cli/cli/layers.py:115]: @MattWhelan I moved the subscription filter code to the same place where we check if the extension is enabled and add the license key policy. Trying to keep the logic in the CLI bits to a minimum.

Comment (kolanos) [reply newrelic_lambda_cli/cli/layers.py:115]: Code was moved here: https://github.com/newrelic/newrelic-lambda-cli/pull/139/files#diff-d14719dcdcf446bd00c4b7977b0acee473aa1e5dbce20b230ccca5e856e3c9cfR191

**Pull request merged**